MixItUp.Base/ViewModel/User/UserViewModel.cs

```diff
@@ -470,9 +398,66 @@ public int CompareTo(UserViewModel other)
 
         private void SetMixerRoles(string[] userRoles)
         {
-            if (userRoles != null && userRoles.Count() > 0)
+            lock (this.mixerRoles)
             {
-                this.chatRoles = new List<string>(userRoles);
+                this.mixerRoles.Clear();
+                this.mixerRoles.Add(MixerRoleEnum.User);
+
+                if (userRoles.Any(r => r.Equals("Owner"))) { this.mixerRoles.Add(MixerRoleEnum.Streamer); }
+                if (userRoles.Any(r => r.Equals("Staff"))) { this.mixerRoles.Add(MixerRoleEnum.Staff); }
+                if (userRoles.Any(r => r.Equals("ChannelEditor"))) { this.mixerRoles.Add(MixerRoleEnum.ChannelEditor); }
+                if (userRoles.Any(r => r.Equals("Mod"))) { this.mixerRoles.Add(MixerRoleEnum.Mod); }
+                if (userRoles.Any(r => r.Equals("Subscriber"))) { this.mixerRoles.Add(MixerRoleEnum.Subscriber); }
+                if (userRoles.Any(r => r.Equals("Partner"))) { this.mixerRoles.Add(MixerRoleEnum.Partner); }
+                if (userRoles.Any(r => r.Equals("Pro"))) { this.mixerRoles.Add(MixerRoleEnum.Pro); }
+                if (userRoles.Any(r => r.Equals("Banned"))) { this.mixerRoles.Add(MixerRoleEnum.Banned); }
+
+                if (ChannelSession.Channel.user.username.Equals(this.UserName))
+                {
+                    this.mixerRoles.Add(MixerRoleEnum.Streamer);
+                }
+
+                if (this.mixerRoles.Contains(MixerRoleEnum.Streamer))
+                {
+                    this.mixerRoles.Add(MixerRoleEnum.Subscriber);
+                    this.mixerRoles.Add(MixerRoleEnum.Mod);
+                    this.mixerRoles.Add(MixerRoleEnum.Follower);
+                }
+
+                if (this.FollowDate != null && this.FollowDate.GetValueOrDefault() > DateTimeOffset.MinValue)
```

Comment:
One issue with this is if the user's chat roles are never updated and we fail on a follower check, the follower role will never get added for the user. Consider moving this up into to around Line 141 in the MixerRoles getter. It will add back a little over-head, but will help cover some of those corner case scenarios.

Reply:
Just pushed an update to address this.